include/eve/algo/concepts.hpp

```diff
@@ -0,0 +1,19 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <concepts>
+
+namespace eve::algo
+{
+  template <typename T>
```

Comment:
I like when we have both the `_t` and the normal traits. Unless this one is for implementation only, maybe we can add the missing one? If it's internal, then this is fine.

Reply:
Had to do like this:

```
  template <typename T>
  using unaligned_t = decltype(std::declval<T>().unaligned());

  template <typename T>
  struct unaligned
  {
    using type = unaligned_t<T>;
  };
```
Otherwise the compilation breaks for some bizarre reason.

Reply:
This is fine